### PR TITLE
COCICCS-6191:AWS roCOCICCS-6191:AWS role policies of Safe, Cert, AD and IAM service account got removed when edited AAD AWS permission via UI

### PR DESCRIPTION
**Pull request by msivatmob**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java [hunk 1]
```diff
@@ -105,6 +106,15 @@
...
 	 */
 	public ResponseEntity<String> createRole(String token, AWSLoginRole awsLoginRole, UserDetails userDetails) throws TVaultValidationException{
+		if(!StringUtils.isEmpty(awsLoginRole.getPolicies())) {
+			logger.debug(JSONUtil.getJSON(ImmutableMap.<String, String>builder().
+					put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER)).
+					put(LogMessage.ACTION, "Checking whether policy is added as an input param for create AWS role").
+					put(LogMessage.MESSAGE, String.format("Trying to create AWS Role [%s]", awsLoginRole.getRole())).
+					put(LogMessage.APIURL, ThreadLocalContext.getCurrentMap().get(LogMessage.APIURL)).
+					build()));
+			throw new TVaultValidationException("Invalid inputs for the given AWS login type.policy is not an input parameter");
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java:116]: Can we updated message to something like, policies are not permitted during creation

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSIAMAuthService.java [hunk 2]
```diff
@@ -73,8 +74,16 @@
...
 	 */
 	public ResponseEntity<String> createIAMRole(AWSIAMRole awsiamRole, String token, UserDetails userDetails) throws TVaultValidationException{
+		if(!StringUtils.isEmpty(awsiamRole.getPolicies())) {
+			logger.debug(JSONUtil.getJSON(ImmutableMap.<String, String>builder().
+					put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER).toString()).
+					put(LogMessage.ACTION, "Checking whether policy is added as an input param for creating AWS IAM role").
+					put(LogMessage.MESSAGE, String.format("Trying to create AWS IAM Role [%s]", awsiamRole.getRole())).
+					put(LogMessage.APIURL, ThreadLocalContext.getCurrentMap().get(LogMessage.APIURL).toString()).
+					build()));
+			throw new TVaultValidationException("Invalid inputs for the given aws login type.policy is not an input parameter");
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSIAMAuthService.java:84]: can we update message

**Review by sarathmohan4 — CHANGES_REQUESTED** [review]

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java (hunk 1 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSAuthService.java]: Message updated

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSIAMAuthService.java (hunk 2 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/AWSIAMAuthService.java]: updated the message

File: tvaultapi/src/main/resources/swaggernotes.properties [hunk 3]
```diff
@@ -372,12 +367,13 @@ SelfSupportController.deleteSafe.notes=<b>Delete Safe and its associated users,
 <b>Three safe types can be deleted: </b>Users safe, shared safe and application safe.<br>
 
 SelfSupportController.updateSafe.value=To update Safe and its associated users, groups, roles
-SelfSupportController.updateSafe.notes=<b>This updates Safe and its associated users, groups, roles.</b><br>\
-<b>path: </b>Path cannot be null or empty. This field provide the path of the safe where the safe can be updated. Example :- users/testSafe<br>\
-<b>Three safe types can be updated: Users safe, shared safe and application safe.<br>\
-<b>name: </b>Name cannot be null or empty. Minimum 3 characters.Only allowed 20 safes for each safe type: users, shared and application.<br>\
-<b>owner: </b>Owner email cannot be null or empty.Should be a valid email.<br>\
-<b>description: </b>Description cannot be null or empty.Minimum 10 characters.<br>
+SelfSupportController.updateSafe.notes=<b>Updates Safe and its associated users, groups, roles.</b><br>\
```
Comment (alexrajan3) [inline tvaultapi/src/main/resources/swaggernotes.properties:370]: Change "Updates" to "Update"

**Review by alexrajan3 — CHANGES_REQUESTED** [review]

File: tvaultapi/src/main/resources/swaggernotes.properties (hunk 3 above)
Comment (msivatmob) [reply tvaultapi/src/main/resources/swaggernotes.properties]: Changed.

**Pull request merged**